app/models/setting.rb

```diff
@@ -296,7 +296,17 @@ def self.column_check(opts)
   # End methods for loading default settings
 
   def full_name_with_default
-    _("%{full_name} (Default: %{default})") % {full_name: _(full_name), default: default}
+    _("%{full_name} (Default: %{default})") % {full_name: _(full_name), default: has_default? ? default : "Not set" }
+  end
+
+  def has_default?
+    case self.settings_type
+    when "array", "string", "hash"
+      !self.default.empty?
+    else
+      # types integer and boolean will always have a default: 0, false
```

Comment:
Nitpick: boolean default value can be either true or false, depending on what you define.